Storage/Bookmarks/BookmarksModel.swift

```diff
@@ -106,6 +110,14 @@ public class BookmarksModel: BookmarksModelFactorySource {
     }
 
     /**
+     * Produce a new model with a memory-backed root with the given GUID removed from the current folder
+     */
+    public func removeGUIDFromCurrent(guid: GUID) -> BookmarksModel {
+        let memoryBackedFolder = MemoryBookmarkFolder.fromBookmarkFolder(self.current)
```

Comment:
Note that using `filter` in this way might even allow Swift's copy-on-write struct ownership analysis to turn this into a single mutation. Neat.